File: L018/minOfArr.cpp
#include<iostream>
#include<vector>
using namespace std;

int minArr(vector<int>& arr, int idx){

    if(idx == arr.size()) return (int)1e9;

    int minEle = minArr(arr,idx+1);
    return min(minEle,arr[idx]);
}

int main(){

    int n;
    cin>>n;

    vector<int>arr(n,0);
    for(int i=0;i<arr.size();i++){
        cin>>arr[i];
    }

    cout<<minArr(arr,0);
    return 0;
}

File: L018/minOfArr.java
import java.util.Scanner;

public class minOfArr {

    public static Scanner sc = new Scanner(System.in);
    public static int minArr(int[] arr, int idx) {

        if (idx == arr.length) return (int)1e9;

        int minEle = minArr(arr, idx + 1);
        return Math.min(arr[idx], minEle);


    }

    public static void main(String[] args) {

        int n = sc.nextInt();

        int[] arr = new int[n];
        for (int i = 0; i < arr.length; i++) {
            arr[i] = sc.nextInt();
        }

        System.out.print(minArr(arr, 0));
    }
}